src/components/tl/nccl/tl_nccl_context.c

```diff
@@ -11,10 +11,37 @@
 #include "core/ucc_ee.h"
 #include "utils/arch/cpu.h"
 
+#if NCCL_USE_NON_BLOCKING
+static ucc_status_t ucc_tl_nccl_nb_progress(ucc_tl_nccl_task_t *task) {
```

Comment:
I think it would be better to do something in the spirit of:
```
static ucc_status_t ucc_tl_nccl_progress(ucc_tl_nccl_task_t *task) {
#if NCCL_USE_NON_BLOCKING
   return ucc_tl_nccl_nb_progress(ucc_tl_nccl_task_t *task);
#else
    return UCC_OK;
#endif
}
```
This way you can avoid many branches, here on line 14 but also each time you call ucc_tl_nccl_nb_progress